js/ad_manager_ooyala.js

```diff
@@ -0,0 +1,207 @@
+     * @public
+     * @param {object} adData Ad-specific metadata
+     */
+    this.ooyalaAdData = function(adData) {
+      if (!adData) return;
+      this.streams[OO.VIDEO.ENCODING.MP4] = adData;
+      this.ready = true;
+      _amc.onAdManagerReady();
+    };
+
+    /**
+     * Called once per video by Ad Manager Controller once the ad manager has set its ready flag to true.
+     * This function asks the ad manager to return a list of all ads to the controller for addition in the
+     * line. If the list of ads is not available at this time, return [] or null and call
+     * [AdManagerController].appendToTimeline() when the ads become available.
+     * The duration and position of each ad should be specified in seconds.
+     * @method AdManager#buildTimeline
+     * @public
+     * @returns {OO.AdManagerController#Ad[]} timeline A list of the ads to play for the current video
+     */
+    this.buildTimeline = function() {
+      var adsTimeline = [];
+      var adData = {
+        "position": this.movieMetadata.ads[0].time,
+        "adManager": this.name,
+        "ad": this.movieMetadata.ads[0],
+        "streams": this.streams,
+        "adType": _amc.ADTYPE.UNKNOWN_AD_REQUEST
+      };
+      var adToInsert = new _amc.Ad(adData);
+      adsTimeline.push(adToInsert);  
+      return adsTimeline;       
+    };
+
+    /**
+     * Called by Ad Manager Controller. The ad manager should play the ad or group of podded ads passed to
+     * the function as a parameter.
+     * @method AdManager#playAd
+     * @public
+     * @param {object} ad The ad object to play
+     */
+    this.playAd = function(ad) {
+      var ooyalaAdAutoPlay = {"autoPlay" : true}; 
+      _adProperties.indexInPod = 1;
+      _adProperties.name = this.movieMetadata.title;
+      _adProperties.duration = this.movieMetadata.duration;
+      _adProperties.hasClickUrl = ad.ad.click_url;
+      _adProperties.skippable = false;
+      _amc.notify(_amc.EVENTS.SET_EMBED_CODE, _adEmbedCode, ooyalaAdAutoPlay);
+      _amc.hidePlayerUi();
+      _amc.notifyPodStarted(ad.id, 1);
+      _amc.notifyLinearAdStarted(ad.id, _adProperties);
+    };
+
+    /**
+     * Called by Ad Manager Controller. The ad manager should cancel the ad passed to the function as a
+     * parameter. After cancelling the ad, the ad manager should call the adEndedCallback to indicate that
+     * ad cancellation has completed. If the given ad is not currently playing and the adEndedCallback has
+     * already been called, then no action is required.
+     * @method AdManager#cancelAd
+     * @public
+     * @param {object} ad The ad object to cancel
+     * @param {object} params An object containing information about the cancellation. It will include the
+     *                        following fields:
+     *                 code : The _amc.AD_CANCEL_CODE for the cancellation
+     */
+    this.cancelAd = function(ad, params) {
+    };
```

Comment:
Should this plugin be doing anything when the ad is cancelled? Anything to cleanup or states to change?